### PR TITLE
Fabric Cloud DNS module

### PR DESCRIPTION
**Pull request by ludoo**
I was looking for an equivalent of the DNS module we traditionally use in Fabric, and noticed this repository has an old PR with unanswered comment, and no complete support of all Cloud DNS features.

I reworked the Fabric module slightly so that it now supports peering and public zones, which we did not have, and added examples. I will add tests in the future if this is merged, all examples work and we've been using the module this is based on for months with no issues, so it should be pretty stable.

### CODE REVIEW

**Comment by ludoo** [comment]
@drebes ping as agreed

File: README.md [hunk 1]
```diff
@@ -1 +1,105 @@
...
+These sections describe requirements for using this module.
+
+### Software
+
+The following dependencies must be available:
+
+- [Terraform][terraform] v0.11
+- [Terraform Provider for GCP][terraform-provider-gcp] plugin v2.0
+
+### Service Account
```
Comment (aaron-lane) [inline README.md]: This section should be updated from the template content.

File: README.md [hunk 2]
```diff
@@ -1 +1,105 @@
...
+
+User or service account credentials with the following roles must be used to provision the resources of this module:
+
+- Storage Admin: `roles/storage.admin`
+
+The [Project Factory module][project-factory-module] and the
+[IAM module][iam-module] may be used in combination to provision a
+service account with the necessary roles applied.
+
+### APIs
```
Comment (aaron-lane) [inline README.md]: This section should be updated from the template content.

File: examples/forwarding-zone/providers.tf [hunk 3]
```diff
@@ -0,0 +1,3 @@
+provider "google" {}
```
Comment (aaron-lane) [inline examples/forwarding-zone/providers.tf]: Do we know of compatible provider versions which we can document as supported?

**Pull request merged**